lib/utils.pm

```diff
@@ -107,6 +108,10 @@ sub wait_boot {
             die "needle 'grub2' not found";
         }
     }
+    else {
+        # With Xen PV we don't see the graphical bootloader
+        record_soft_failure 'missing implementation';
```

Comment:
IMHO we should reserve **soft_failure** for known product issues. If the test is calling this but missing implementation it will either pass so that we don't need to do anything or it will fail later with a more hideous symptom so better just die here with an explicit message.